invenio_circulation/transitions/base.py

```diff
@@ -105,30 +105,29 @@ def __init__(self, src, dest, trigger='next', permission_factory=None,
     def ensure_item_is_available(self, loan):
         """Validate that an item is available."""
         if 'item_pid' not in loan:
-            msg = 'Item not set for loan'.format(loan['loan_pid'])
-            raise TransitionConstraintsViolation(msg=msg)
+            msg = "Item not set for loan with pid '{}'".format(loan.id)
+            raise TransitionConstraintsViolationError(description=msg)
 
         if not is_item_available(loan['item_pid']):
-            msg = 'Invalid transition to {0}: Item {1} is unavailable.'\
-                .format(self.dest, loan['item_pid'])
-            raise ItemNotAvailable(msg)
+            raise ItemNotAvailableError(
+                item_pid=loan['item_pid'], transition=self.dest
+            )
 
     def validate_transition_states(self):
         """Ensure that source and destination states are valid."""
         states = current_app.config['CIRCULATION_LOAN_TRANSITIONS'].keys()
         if not all([self.src in states, self.dest in states]):
-            msg = 'Source state `{0}` or destination state `{1}` not in [{2}]'\
+            msg = "Source state '{0}' or destination state '{1}' not in [{2}]"\
                 .format(self.src, self.dest, states)
-            raise InvalidState(msg=msg)
+            raise InvalidLoanStateError(description=msg)
 
     @check_trigger
     @ensure_required_params
     @ensure_same_item_patron
     def before(self, loan, **kwargs):
         """Validate input, evaluate conditions and raise if failed."""
         if self.permission_factory and not self.permission_factory(loan).can():
-            msg = 'Invalid circulation permission'
-            raise InvalidCirculationPermission(msg=msg)
+            raise InvalidPermissionError()
```

Comment:
no description?

Reply:
If we don't provide description it will use the default one.